### PR TITLE
Add support for @Small directive

### PR DESCRIPTION
**Pull request by ethan-kusters**
Bug/issue #, if applicable: rdar://97744845

## Summary

Adds support for the `@Small` directive as described here: https://forums.swift.org/t/supporting-more-dynamic-content-in-swift-docc-reference-documentation/59527#small-18

A new `@Small` directive based on HTML’s small tag. This directive can be used to specify small print like legal, license, or copyright text that should be rendered in a smaller font size.

`@Small` accepts no parameters and accepts arbitrary DocC markup body content.

Example:

    You can create a sloth using the ``init(name:color:power:)``
    initializer, or create randomly generated sloth using a
    ``SlothGenerator``:

       let slothGenerator = MySlothGenerator(seed: randomSeed())
       let habitat = Habitat(isHumid: false, isWarm: true)

       // ...

    @Small {
        Licensed under Apache License v2.0 with Runtime Library Exception
    }

> **Note**
> The implementation for `@Small` relies on some new infrastructure added in #378 to support directives that include directly renderable content.

## Dependencies

- https://github.com/apple/swift-docc-render/pull/405

## Testing

Using the [latest version of Swift-DocC-Render](https://github.com/apple/swift-docc-render-artifact) (support for `@Small` landed on the `main` branch recently), build documentation for a DocC catalog that includes the `@Small` directive. Confirm it renders as expected.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran the `./bin/test` script and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

File: Sources/SwiftDocC/SwiftDocC.docc/Resources/RenderNode.spec.json [hunk 1]
```diff
@@ -619,6 +622,25 @@
                     }
                 }
             },
+            "Small": {
+                "type": "object",
+                "required": [
+                    "type",
+                    "content",
```
Comment (mportiz08) [inline Sources/SwiftDocC/SwiftDocC.docc/Resources/RenderNode.spec.json:629]:
```suggestion
                    "inlineContent",
```

(Or if it's meant to actually be `content`, then I think the actual implementation is using the wrong key—testing this out with Dobri's PR now)

Comment (ethan-kusters) [reply Sources/SwiftDocC/SwiftDocC.docc/Resources/RenderNode.spec.json]: Oh nice catch- thank you!

**Comment by ethan-kusters** [comment]
@swift-ci please test

File: Sources/SwiftDocC/Model/Rendering/Content/RenderBlockContent.swift [hunk 2]
```diff
@@ -61,7 +61,11 @@ public enum RenderBlockContent: Equatable {
     /// A table that contains a list of row data.
     case table(Table)
     
+    /// A row in a grid-based layout system that describes a collection of columns.
```
[the remark this answers is not in the document]
Comment (ethan-kusters) [reply Sources/SwiftDocC/Model/Rendering/Content/RenderBlockContent.swift:64]: Yeah agreed- I just missed it #378 so I thought it was worth cleaning up here. I can split it into a separate commit though.

File: Sources/SwiftDocC/SwiftDocC.docc/Resources/RenderNode.spec.json [hunk 3]
```diff
@@ -619,6 +622,25 @@
                     }
                 }
             },
+            "Small": {
+                "type": "object",
+                "required": [
+                    "type",
+                    "inlineContent",
```
Comment (daniel-grumberg) [inline Sources/SwiftDocC/SwiftDocC.docc/Resources/RenderNode.spec.json:629]: Nit trailing comma

File: Tests/SwiftDocCTests/Rendering/RenderNodeTranslatorTests.swift [hunk 4]
```diff
@@ -1057,5 +1057,15 @@ class RenderNodeTranslatorTests: XCTestCase {
         XCTAssertEqual(row.columns.first?.content.count, 1)
         XCTAssertEqual(row.columns.last?.size, 5)
         XCTAssertEqual(row.columns.last?.content.count, 3)
+        
+        guard case let .small(small) = discussion.content.last else {
```
Comment (daniel-grumberg) [inline Tests/SwiftDocCTests/Rendering/RenderNodeTranslatorTests.swift:1061]: Should we not have a separate test from row and column one for this? I find that having smaller test cases helps pin out failures faster.

File: Tests/SwiftDocCTests/Semantics/Reference/RowTests.swift [hunk 5]
```diff
@@ -29,8 +29,9 @@ class RowTests: XCTestCase {
             ["1: warning – org.swift.docc.HasAtLeastOne<Row, Column>"]
         )
         
+        XCTAssertEqual(renderBlockContent.count, 1)
         XCTAssertEqual(
-            renderBlockContent,
+            renderBlockContent.first,
```
Comment (daniel-grumberg) [inline Tests/SwiftDocCTests/Semantics/Reference/RowTests.swift:34]: Should this not be in a separate patch?

File: Sources/SwiftDocC/Model/Rendering/Content/RenderBlockContent.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (daniel-grumberg) [reply Sources/SwiftDocC/Model/Rendering/Content/RenderBlockContent.swift:64]: That would be nice

**Comment by daniel-grumberg** [comment]
Looks pretty good to me, small fixes and then LGTM

File: Sources/SwiftDocC/Model/Rendering/Content/RenderBlockContent.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (ethan-kusters) [reply Sources/SwiftDocC/Model/Rendering/Content/RenderBlockContent.swift:64]: Split out here: e56bd43357608afcfd5e9edfa04ed88191bbb474

File: Tests/SwiftDocCTests/Rendering/RenderNodeTranslatorTests.swift (hunk 4 above)
Comment (ethan-kusters) [reply Tests/SwiftDocCTests/Rendering/RenderNodeTranslatorTests.swift:1084]: Ah good catch- I meant to create a new one here. Should be fixed now. Thanks!

File: Tests/SwiftDocCTests/Semantics/Reference/RowTests.swift (hunk 5 above)
Comment (ethan-kusters) [reply Tests/SwiftDocCTests/Semantics/Reference/RowTests.swift:34]: Split into a separate commit here: e4a6fc4041d7a6a1d18733aa3d3b564da7fd58c3.

**Comment by ethan-kusters** [comment]
@swift-ci please test

**Review by daniel-grumberg — APPROVED** [review]
LGTM
